whetstone/compiler/src/main/kotlin/com/freeletics/mad/whetstone/WhetstoneCodeGenerator.kt

```diff
@@ -158,6 +164,36 @@ class WhetstoneCodeGenerator : CodeGenerator {
         )
     }
 
+    private fun generateComposeDialogFragmentCode(
+        codeGenDir: File,
+        module: ModuleDescriptor,
+        declaration: KtDeclaration
+    ): GeneratedFile? {
+        val composeFragment = declaration.findAnnotation(composeDialogFragmentFqName, module) ?: return null
+        val data = ComposeFragmentData(
+            baseName = declaration.name!!,
+            packageName = declaration.containingKtFile.packageFqName.pathSegments().joinToString(separator = "."),
+            scope = composeFragment.requireClassArgument("scope", 0, module),
+            parentScope = composeFragment.requireClassArgument("parentScope", 1, module),
+            dependencies = composeFragment.requireClassArgument("dependencies", 2, module),
+            fragmentBaseClass = composeFragment.requireClassArgument("dialogFragmentBaseClass", 3, module),
```

Comment:
I haven't found a way so far to read the default value of an annotation, so for now you need too use `optionalClassArgument` and add `?: dialogFragment`